docs/xml-sitemaps.md

```diff
@@ -2,70 +2,164 @@
+
+	/**
+	 * Set the name and object type properties. Required.
+	 */
+	public function __construct() {
+		$this->name = 'videos';
+		$this->object_type = 'post';
 	}
 
-	$url->changefreq = new SimpleXMLElement( '<changefreq>always</changefreq>' );
-	$url->priority = new SimpleXMLElement( '<priority>1.0</priority>' );
+	/**
+	 * Return the list of URLs for the current page.
+	 */
+	public function get_url_list( $page ) : array {
+		$videos = new WP_Query( [
+			'post_type' => 'video',
+			'posts_per_page' => 2000,
+			'fields' => 'ids',
+			'paged' => $page,
+		] );
 
-	return $url;
-} );
-```
+		$urls = [];
 
-You can use this filter to add more detailed information to enrich your sitemaps with [Video](https://support.google.com/webmasters/answer/80471), [Image](https://support.google.com/webmasters/answer/178636) and [News](https://support.google.com/webmasters/answer/74288) data.
+		foreach ( $videos->posts as $video_id ) {
+			$urls[] = get_post_meta( $video_id, 'video_url', true );
+		}
 
-## Adding an XML namespace
+		return $urls;
+	}
 
-Some custom extensions to the standard sitemap format are supported but require the addition of one or more XML namespaces for them to validate. This is achieved using the `msm_sitemap_namespace` filter. The below example adds Google's video sitemap schema.
+	/**
+	 * Return the maximum number of pages to output for this sitemap.
+	 */
+	public function get_max_num_pages() {
+		$videos = new WP_Query( [
+			'post_type' => 'video',
+			'posts_per_page' => 2000,
+			'fields' => 'ids',
+		] );
+
+		return $videos->max_num_pages;
+	}
 
-```php
-add_filter( 'msm_sitemap_namespace', function ( array $namespaces ) : array {
-	$namespaces['xmlns:video'] = "http://www.google.com/schemas/sitemap-video/1.1";
-	return $namespaces;
+}
+
+add_filter( 'init', function() {
+	$provider = new Video_Sitemap_Provider();
+	wp_register_sitemap_provider( 'video-sitemaps', $provider );
 } );
 ```
 
-## Skipping a post
+The provider will be responsible for getting all sitemaps and sitemap entries, as well as determining pagination.
+
+## Removing Specific Sitemaps
+There are three existing sitemaps providers for standard object types - `posts`, `taxonomies`, and `users`. If you want to remove one of them such as the "users" provider, you can configure the module like so:
+
+```json
+{
+	"extra": {
+		"altis": {
+			"modules": {
+				"seo": {
+					"xml-sitemaps": {
+						"users": false
+					}
+				}
+			}
+		}
+	}
+}
+```
 
-The `msm_sitemap_skip_post` filter is used to exclude posts from the sitemaps programmatically. The current post is available via `get_post()`.
+Alternatively if you need to use more complex logic or work with custom sitemap providers you can use the `wp_sitemaps_add_provider` filter to do so for example:
 
 ```php
-add_filter( 'msm_sitemap_skip_post', function () {
-	if ( get_post_type() === 'hideme' ) {
-		return true;
+add_filter( 'wp_sitemaps_add_provider', function( $provider, $name ) {
```

Comment:
Do we need to document how to migrate these old filters?

Reply:
🤔 we don't need to per se but we could do to be nice. I'd probably do that on the upgrade guide though